src/backend/marsha/core/permissions.py

```diff
@@ -24,6 +25,28 @@ def has_permission(self, request, view):
         return False
 
 
+class UserOrResourceIsAuthenticated(IsAuthenticated):
+    """This allows access for in user or resource context."""
+
+
+class UserIsAuthenticated(IsAuthenticated):
+    """This allows access only in user context."""
+
+    def has_permission(self, request, view):
+        """Simply checks we are NOT in a resource context."""
+        has_permission = super().has_permission(request, view)
+        return has_permission and request.resource is None
+
+
+class ResourceIsAuthenticated(IsAuthenticated):
```

Comment:
This class name seems strange to me…
What about something like `HasAccessToResource` or `IsResourceToken` ?

Reply:
As discussed, we will keep `ResourceIsAuthenticated` as a name, even if it's odd to have an "authenticated resource", at least it keeps the naming explicit, according to DRF (`IsAuthenticated`) and the new `request.resource`.